datafusion/core/src/datasource/avro_to_arrow/arrow_array_reader.rs

```diff
@@ -97,15 +97,15 @@ impl<'a, R: Read> AvroArrowArrayReader<'a, R> {
         schema_lookup: &'b mut BTreeMap<String, usize>,
     ) -> Result<&'b BTreeMap<String, usize>> {
         match schema {
-            AvroSchema::Record(RecordSchema {
-                name,
-                fields,
-                lookup,
-                ..
-            }) => {
+            AvroSchema::Union(union_schema) => {
+                if union_schema.is_nullable() {
+                    let rec_schema = &union_schema.variants()[1];
```

Comment:
This part seems incorrect. The second variants is not always non-null, but the first one can be.

Reply:
Also, it's better to check if the number of variants is 2 because this logic only consider nullable types, not general union types which contains 3 or more variants.